### PR TITLE
feat: handle official user-moderation-notifications pubsub topic

### PR DESCRIPTION
**Pull request by iProdigy**
### Prerequisites for Code Changes
* [x] This pull request follows the code style of the project
* [ ] I have tested this feature

### Changes Proposed
* Implement official `user-moderation-notifications.<user-id>.<channel-id>`  pubsub topic

### Additional Information
Depends on #331

### CODE REVIEW

File: pubsub/src/main/java/com/github/twitch4j/pubsub/TwitchPubSub.java [hunk 1]
```diff
@@ -564,6 +564,16 @@ public void onTextMessage(WebSocket ws, String text) {
                                         log.warn("Unparsable Message: " + message.getType() + "|" + message.getData());
                                         break;
                                 }
+                            } else if (topic.startsWith("user-moderation-notifications")) {
+                                int lastPeriod = topic.lastIndexOf('.');
+                                String userId = topic.substring(topic.indexOf('.') + 1, lastPeriod);
```
Comment (PhilippHeuer) [inline pubsub/src/main/java/com/github/twitch4j/pubsub/TwitchPubSub.java:569]: Maybe we can improve this a little by adding sth. after https://github.com/twitch4j/twitch4j/blob/39df41f85ce6cdc6810c4cdeb04a347aa0bbb1e4/pubsub/src/main/java/com/github/twitch4j/pubsub/TwitchPubSub.java#L369 to parse the topic value first (split on `.`).

And then check for a exact match on the topic name and check that the param count (ie. 3 for this one, topic + userId + channelId) matches.
We wouldn't need the lastIndexOf / substring code in every case then.

**Review by PhilippHeuer — CHANGES_REQUESTED** [review]

File: pubsub/src/main/java/com/github/twitch4j/pubsub/TwitchPubSub.java (hunk 1 above)
Comment (iProdigy) [reply pubsub/src/main/java/com/github/twitch4j/pubsub/TwitchPubSub.java]: Good idea, I switched to the split approach for this topic, will migrate the others in a separate refactor PR

File: pubsub/src/main/java/com/github/twitch4j/pubsub/domain/AutomodContentClassification.java [hunk 2]
```diff
@@ -8,11 +9,50 @@
...
+        // New Categories
+        AGGRESSION,
+        BULLYING,
+        DISABILITY,
+        MISOGYNY,
+        RACE_ETHNICITY_OR_RELIGION,
+        SEXUALITY_SEX_OR_GENDER,
+        SEX_BASED_TERMS,
+        SWEARING,
+        // Possible Categories
```
Comment (iProdigy) [inline pubsub/src/main/java/com/github/twitch4j/pubsub/domain/AutomodContentClassification.java:32]: Could do some manual parsing work here to consolidate enum values

File: pubsub/src/main/java/com/github/twitch4j/pubsub/domain/AutomodCaughtMessage.java [hunk 3]
```diff
@@ -35,15 +63,37 @@
     @Setter(AccessLevel.PRIVATE)
     public static class FragmentFlags {
         private Map<String, Integer> topics;
+
+        public Map<AutomodContentClassification.Category, Integer> getParsedTopics() {
+            return TypeConvert.convertValue(topics, new TypeReference<EnumMap<AutomodContentClassification.Category, Integer>>() {});
```
Comment (iProdigy) [inline pubsub/src/main/java/com/github/twitch4j/pubsub/domain/AutomodCaughtMessage.java:68]: technically not the most performant solution, but this supports `@JsonAlias`

**Review by PhilippHeuer — APPROVED** [review]

**Pull request merged**